### PR TITLE
CLOUDP-114080: [Mongocli] Investigate how to get the correct git tag version in evergreen

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-114080

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments
Issue: We have several evergreen tasks that are using git describe to get the latest tag version available. Git describe will get you the latest tag, for now it works since we only release one thing, this may become flaky later. This PR address this comment https://github.com/mongodb/mongocli/pull/1008#discussion_r816713273

**Nre Version format for non-production binaries:** `vX.Y.Z-next`.

```
make build
==> Building mongocli binary
go build -ldflags "-s -w -X github.com/mongodb/mongocli/internal/version.GitCommit=cf0d9451ce3aec441493053a48c9d18286b12bd2 -X github.com/mongodb/mongocli/internal/config.ToolName=mongocli -X github.com/mongodb/mongocli/internal/version.Version=1.23.0-next" -o ./bin/mongocli ./cmd/mongocli
```

```
./bin/mongocli --version
mongocli version: 1.23.0-next
git version: cf0d9451ce3aec441493053a48c9d18286b12bd2
Go version: go1.17
   os: darwin
   arch: amd64
   compiler: gc
```

We use `git describe` if the tag is not found.
```
make build-atlascli
==> Building atlas binary
go build -ldflags "-s -w -X github.com/mongodb/mongocli/internal/version.GitCommit=11f126fcf02994e500b73bfa4a9b437fff8c7415 -X github.com/mongodb/mongocli/internal/config.ToolName=atlascli -X github.com/mongodb/mongocli/internal/version.Version=1.23.0-24-g11f126fc" -o ./bin/atlas ./cmd/atlas
```

```
➜  ~/workspace/mongocli git:(CLOUDP-114080) ./bin/atlas --version
atlascli version: 1.23.0-24-g11f126fc
git version: 11f126fcf02994e500b73bfa4a9b437fff8c7415
Go version: go1.17
   os: darwin
   arch: amd64
   compiler: gc
```

I have tested my fix using this fork https://github.com/andreaangiolillo/mongocli-test/releases which contains atlascli tags

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,12 +16,13 @@ MCLI_E2E_BINARY?=../../bin/${MCLI_BINARY_NAME}
 
 ATLAS_SOURCE_FILES?=./cmd/atlas
 ATLAS_BINARY_NAME=atlas
+ATLAS_VERSION?=$(shell git tag --list 'atlascli/v*' --sort=committerdate | tail -1 | cut -d "v" -f 2 | xargs -I % sh -c 'echo %-next' )
```
Comment (andreaangiolillo) [inline Makefile:19]: This piece of code and the evergreen packaging tasks won’t work until we have an `atlascli/v*` tag in the repository

File: .atlascli.goreleaser.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ builds:
       binary: bin/atlas
       main: ./cmd/atlas/atlas.go
       ldflags:
-        - -s -w -X github.com/mongodb/mongocli/internal/version.Version={{.Version}} -X github.com/mongodb/mongocli/internal/version.GitCommit={{.FullCommit}}
+        - -s -w -X github.com/mongodb/mongocli/internal/version.Version={{ .Env.VERSION_GIT }} -X github.com/mongodb/mongocli/internal/version.GitCommit={{.FullCommit}}
```
Comment (gssbzn) [inline .atlascli.goreleaser.yml:17]: why is this needed? go releaser should be handling this correctly if they don't let's open an issue for them

File: .mongocli.goreleaser.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ builds:
       binary: bin/mongocli
       main: ./cmd/mongocli/mongocli.go
       ldflags:
-        - -s -w -X github.com/mongodb/mongocli/internal/version.Version={{.Version}} -X github.com/mongodb/mongocli/internal/version.GitCommit={{.FullCommit}}
+        - -s -w -X github.com/mongodb/mongocli/internal/version.Version={{ .Env.VERSION_GIT }} -X github.com/mongodb/mongocli/internal/version.GitCommit={{.FullCommit}}
```
Comment (gssbzn) [inline .mongocli.goreleaser.yml:17]: same as above

**Review by gssbzn — COMMENTED** [review]
missing this one https://github.com/mongodb/mongocli/blob/master/build/package/generate-msi.sh

probably a good idea to do a search/grep for `git describe` in the whole repo to make sure we don't miss one

File: build/package/generate-download-archive-manifest.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+if [[ -z "${VERSION}" ]]; then
+    VERSION=$(git describe | cut -d "v" -f 2)
+fi
+
+go run ../internal/release/main.go --file "${FEED_FILE_NAME}" --version "${VERSION}"
+
+
+
+
+
```
Comment (gssbzn) [inline build/package/generate-download-archive-manifest.sh:29]:
```suggestion
go run ../internal/release/main.go --file "${FEED_FILE_NAME}" --version "${VERSION}"
```

File: .atlascli.goreleaser.yml (hunk 2 above)
Comment (andreaangiolillo) [reply .atlascli.goreleaser.yml:17]: yes, you are right. Thanks!

**Review by gssbzn — APPROVED** [review]
LGTM and godspeed with your first atlas alpha
